### PR TITLE
Enable live tests in non ci

### PR DESCRIPTION
**Pull request by AlexGhiondea**
## Description

Enable the AzConfig tests against the live service. They should not run during CI.

/cc @maririos @mikeharder @weshaggard @bsiegel

### CODE REVIEW

File: .azure-pipelines/client.yml [hunk 1]
```diff
@@ -39,7 +39,18 @@ jobs:
...
+        inputs:
+          command: test
+          projects: '$(ProjectFile)'
+          arguments: --filter TestCategory!=Live
+
+      - task: DotNetCoreCLI@2
+        displayName: 'Build & Test (with live tests)'
+        condition: ne(variables['System.TeamProject'], 'public')
+        env:
+          AZ_CONFIG_CONNECTION: $(AzConfigConnectionString)
```
Comment (maririos) [inline .azure-pipelines/client.yml]: The pipeline needs to know where to get the `AzConfigConnectionString` value from. We will need to add something like: https://github.com/Azure/azure-sdk-for-net-lab/blob/master/.azure-pipelines/azure-pipelines.yml#L1
and add the corresponding key from KV tot he internal pipeline definition

File: .azure-pipelines/client.yml [hunk 2]
```diff
@@ -39,7 +39,18 @@ jobs:
           version: 2.1.503
 
       - task: DotNetCoreCLI@2
-        displayName: 'Build & Test'
+        displayName: 'Build & Test (no live tests)'
```
Comment (maririos) [inline .azure-pipelines/client.yml]: Looks like this was only added for the Windows leg. We will need to do the same in the Linux one

Comment (AlexGhiondea) [reply .azure-pipelines/client.yml]: Indeed -- I didn't page down in the file 🤦‍♂️

**Comment by mikeharder** [comment]
This change will have merge conflicts with https://github.com/Azure/azure-sdk-for-net/pull/5233/files.  I think 5233 should be merged first.  @KarishmaGhiya, can you merge this soon?

**Review by maririos — APPROVED** [review]
We need to make sure that the build has the connection to the group `AzureSDK KeyVault - AzConfig LiveTest` and that the variable `AzConfigConnection` is available so this change doesn't break internal builds

File: .azure-pipelines/client.yml [hunk 3]
```diff
@@ -1,9 +1,19 @@
...
+variables:
 # ProjectFile - The project to build and test. This variable is defined in pipeline web ui because we want to be able to provide it at queue time and that isn't supported in yaml yet.
 
+ # public
+ - ${{ if eq(variables['System.TeamProject'], 'public') }}:
+   - name: AzConfigConnectionString
+     value: ''
+ # internal
+ - ${{ if ne(variables['System.TeamProject'], 'public') }}:
+   - group: AzureSDK KeyVault - AzConfig LiveTest
```
Comment (weshaggard) [inline .azure-pipelines/client.yml]: As part of our [guidelines](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki?wikiVersion=GBwikiMaster&pagePath=%2FAzure%20DevOps%20Pipeline%20Guidance&pageId=2&anchor=variable-groups) we shouldn't include the group directly in the yaml file we should instead just link it in the build pipeline.

Comment (AlexGhiondea) [reply .azure-pipelines/client.yml]: Currently the pipeline is not setup for this. I'll leave this as-is for now and file an issue to conform to the guidelines -- would that work?

Comment (weshaggard) [reply .azure-pipelines/client.yml]: I would just remove this as when you setup the pipeline you have the link the variable group regardless to make it work.

Comment (AlexGhiondea) [reply .azure-pipelines/client.yml]: Are you saying to remove the line with the group?

Comment (weshaggard) [reply .azure-pipelines/client.yml]: Exactly all you should need to do is delete the one line with `-group ...`.

File: .azure-pipelines/client.yml [hunk 4]
```diff
@@ -1,9 +1,18 @@
 trigger:
 - master
 
-# variables:
+variables:
 # ProjectFile - The project to build and test. This variable is defined in pipeline web ui because we want to be able to provide it at queue time and that isn't supported in yaml yet.
 
+ # public
```
Comment (weshaggard) [inline .azure-pipelines/client.yml]: Not blocking but I thinking about this a little more we should probably just delete this entire section. Don't worry about doing it for your PR unless you want to as I will clean this as well as the net-lab repo up.

Comment (AlexGhiondea) [reply .azure-pipelines/client.yml]: I will take care of that as well.

File: .azure-pipelines/client.yml [hunk 5]
```diff
@@ -66,7 +67,18 @@ jobs:
...
-        displayName: 'Build & Test'
+        displayName: 'Build & Test (no live tests)'
+        condition: eq(variables['System.TeamProject'], 'public')
         inputs:
           command: test
           projects: '$(ProjectFile)'
+          arguments: --filter TestCategory!=Live
+        
+      - task: DotNetCoreCLI@2
+        displayName: 'Build & Test (with live tests)'
```
Comment (weshaggard) [inline .azure-pipelines/client.yml]: Just for clarification this runs all tests correct? Or does it only run live tests? For example does it also run the recorded tests?

**Pull request merged**